### PR TITLE
[TC-5746] iOS: Added optionsImages in OptionDialog

### PR DESCRIPTION
**Pull request by kopiro**
https://jira.appcelerator.org/browse/TC-5746

### CODE REVIEW

**Comment by hansemannn** [comment]
Hi @kopiro, also thanks for this PR! Please provide a sample code before we do a further review. Also check the removed (red) lines again.

File: iphone/Classes/TiUIOptionDialogProxy.m [hunk 1]
```diff
@@ -99,17 +101,22 @@ -(void)show:(id)args
...
             UIAlertAction* theAction = [UIAlertAction actionWithTitle:@"Cancel"
                                                                 style:UIAlertActionStyleCancel
                                                               handler:^(UIAlertAction * action){
                                                                   [self fireClickEventWithAction:action];
                                                               }];
+            if (!IS_NULL_OR_NIL(imagesOptions)) {
+                UIImage *image = [ [UIImage imageNamed:[imagesOptions objectAtIndex:curIndex]] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
+                [theAction setValue:image forKey:@"image"];
+            }
+
```
Comment (hansemannn) [inline iphone/Classes/TiUIOptionDialogProxy.m]: This code would only be executed, if the current device is an iPad and no ``cancel`` property is set. It also does not check for array bounds (use case: options = ["test1","test2"] and imageOptions = ["image.png"]). So:
- Rename ``imagesOptions`` to ``imageOptions``
- Check the array bounds
- Use the TiUtils helper to create an image. It makes it save for the new asset catalog coming by default in 5.2.0.

Example:
```
NSMutableArray *imagesOptions = [self valueForKey:@"imageOptions"];

...
 
if (imagesOptions != nil && curIndex < [imagesOptions count]) {
  UIImage *image = [ [TiUtils image:[imagesOptions objectAtIndex:curIndex] proxy:self] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal];
  [theAction setValue:image forKey:@"image"];
}
```

Comment (kopiro) [inline iphone/Classes/TiUIOptionDialogProxy.m]: @hansemannn I tried to understand TiHelpers, can you provide me sample / doc to create images in a right way?
Thanks!

**Comment by hansemannn** [comment]
Closing this PR for now after discussing with @kopiro, that we should wrap the image options as part of a refactoring that includes an object instead of string values inside the ``options`` property.

**Pull request closed without merging**